service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -78,120 +89,163 @@ private fun Database.Read.getUnitOperationalDays(unitId: DaycareId, dateRange: F
+    """.trimIndent()
+)
+    .bind("childIds", childIds.toTypedArray())
+    .map { row -> toDailyServiceTimes(row) }
+    .toList()
+    .filterNotNull()
+    .associateBy { it.childId }
+
+private fun mapChildReservations(rows: List<UnitAttendanceReservations.QueryRow>, serviceTimes: Map<UUID, DailyServiceTimes>): List<UnitAttendanceReservations.ChildReservations> {
+    return rows
+        .groupBy { it.child }
+        .map { (child, rowsByChild) ->
+            UnitAttendanceReservations.ChildReservations(
+                child = child.copy(dailyServiceTimes = serviceTimes.get(child.id)),
+                dailyData = rowsByChild
+                    .associateBy { it.date }
+                    .map { (date, rowByDateAndChild) ->
+                        UnitAttendanceReservations.DailyChildData(
+                            date = date,
+                            reservation = rowByDateAndChild.reservation?.let {
+                                UnitAttendanceReservations.ReservationTimes(
+                                    startTime = it.startTime,
+                                    endTime = it.endTime
+                                )
+                            },
+                            attendance = rowByDateAndChild.attendance?.let {
+                                UnitAttendanceReservations.AttendanceTimes(
+                                    startTime = it.start.toLocalTime().format(DateTimeFormatter.ofPattern("HH:mm")),
+                                    endTime = it.end?.toLocalTime()?.format(DateTimeFormatter.ofPattern("HH:mm")),
+                                )
+                            },
+                            absence = rowByDateAndChild.absence?.let {
+                                UnitAttendanceReservations.Absence(
+                                    type = it.type
+                                )
+                            }
+                        )
+                    }
+                    .sortedBy { it.date }
+            )
+        }
+        .sortedBy { "${it.child.firstName} ${it.child.lastName}" }
+}
+
 data class UnitAttendanceReservations(
     val unit: String,
     val operationalDays: List<OperationalDay>,
     val groups: List<GroupAttendanceReservations>,
     val ungrouped: List<ChildReservations>
 ) {
+    data class OperationalDay(
+        val date: LocalDate,
+        val isHoliday: Boolean
+    )
+
     data class GroupAttendanceReservations(
         val group: String,
         val children: List<ChildReservations>
     )
 
     data class ChildReservations(
         val child: Child,
-        val reservations: Map<LocalDate, Reservation>
+        val dailyData: List<DailyChildData>
+    )
+
+    data class DailyChildData(
+        val date: LocalDate,
+        val reservation: ReservationTimes?,
+        val attendance: AttendanceTimes?,
+        val absence: Absence?
     )
 
-    data class Reservation(
+    data class ReservationTimes(
```

Comment:
As it is this class is identical to `QueryRowReservation`. Doesn't seem like they are both needed. The same is true for `Absence` and `QueryRowAbsence`.